### PR TITLE
GH-691: Add CompositeProducerListener

### PR DESCRIPTION
**Pull request by garyrussell**
Fixes https://github.com/spring-projects/spring-kafka/issues/691

**cherry-pick to 2.1.x**

### CODE REVIEW

File: spring-kafka/src/main/java/org/springframework/kafka/support/CompositeProducerListener.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		this.delegates.add(delegate);
+	}
+
+	public boolean removeDelegate(ProducerListener<K, V> delegate) {
+		return this.delegates.remove(delegate);
+	}
+
+	@Override
+	public void onSuccess(ProducerRecord<K, V> producerRecord, RecordMetadata recordMetadata) {
+		this.delegates.forEach(d -> d.onSuccess(null, recordMetadata));
```
Comment (artembilan) [inline spring-kafka/src/main/java/org/springframework/kafka/support/CompositeProducerListener.java]: Why `null`? What am I missing?
Thanks

File: spring-kafka/src/test/java/org/springframework/kafka/core/KafkaTemplateTests.java [hunk 2]
```diff
@@ -242,16 +243,19 @@ public void withListener() throws Exception {
...
-		final CountDownLatch latch = new CountDownLatch(1);
-		template.setProducerListener(new ProducerListener<Integer, String>() {
+		final CountDownLatch latch = new CountDownLatch(2);
+		class PL implements ProducerListener<Integer, String> {
 
 			@Override
-			public void onSuccess(String topic, Integer partition, Integer key, String value,
-					RecordMetadata recordMetadata) {
+			public void onSuccess(ProducerRecord<Integer, String> record, RecordMetadata recordMetadata) {
 				latch.countDown();
```
Comment (artembilan) [inline spring-kafka/src/test/java/org/springframework/kafka/core/KafkaTemplateTests.java]: Sounds like test is not full - no `record` and `recordMetadata` assertions.

File: spring-kafka/src/main/java/org/springframework/kafka/support/CompositeProducerListener.java (hunk 1 above)
Comment (garyrussell) [reply spring-kafka/src/main/java/org/springframework/kafka/support/CompositeProducerListener.java]: I have no idea where that came from 😄

**Comment by artembilan** [comment]
... and cherry-picked to `2.1.x`.

**Pull request merged**